src/test/java/org/assertj/core/internal/maps/Maps_assertAnySatisfyingConsumer_Test.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2018 the original author or authors.
+ */
+package org.assertj.core.internal.maps;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.assertj.core.api.Assertions.assertThatNullPointerException;
+import static org.assertj.core.data.MapEntry.entry;
+import static org.assertj.core.error.ElementsShouldSatisfy.elementsShouldSatisfyAny;
+import static org.assertj.core.test.Maps.mapOf;
+import static org.assertj.core.test.TestData.someInfo;
+import static org.assertj.core.test.TestFailures.failBecauseExpectedAssertionErrorWasNotThrown;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+import static org.mockito.Mockito.verify;
+
+import java.util.Map;
+
+import org.assertj.core.internal.MapsBaseTest;
+import org.assertj.core.test.Player;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class Maps_assertAnySatisfyingConsumer_Test extends MapsBaseTest {
+
+  private Map<String, Player> greatPlayers;
+
+  @Override
+  @BeforeEach
+  public void setUp() {
+    super.setUp();
+    greatPlayers = mapOf(entry("Bulls", jordan), entry("Spurs", duncan), entry("Lakers", magic));
+  }
+
+  @Test
+  public void should_pass_if_one_entry_satisfies_the_given_requirements() {
+    maps.assertAnySatisfy(someInfo(), greatPlayers, (team, player) -> {
+      assertThat(team).isIn("Lakers");
+      assertThat(player.getPointsPerGame()).isGreaterThan(18);
+    });
+  }
+
+  @Test
+  public void should_fail_if_the_map_under_test_is_empty_whatever_the_assertions_requirements_are() {
+    actual.clear();
+    try {
+      maps.assertAnySatisfy(someInfo(), actual, ($1,$2) -> assertThat(true).isTrue());
```

Comment:
refactor the test with `AssertionUtil.expectAssertionError` as in https://github.com/joel-costigliola/assertj-core/blob/6e006e39ab86b8ff6322c880627b354ab10b7785/src/test/java/org/assertj/core/internal/objectarrays/ObjectArrays_assertHasOnlyElementsOfTypes_Test.java#L58.

Could you also do that for other `should_fail...` tests ?